Libraries/RhythmGamesBot.Utils/Option.cs

```diff
@@ -51,11 +51,11 @@ public Option<T> Replace(T value)
                 T temp = this._value;
                 this._value = value;
 
-                return temp;
+                return new Option<T>(temp);
```

Comment:
shouldn't it just return `this`?

Reply:
According to the comment, it should return the old value, at least that it how I understand it. 
But my initial though was the same as yours

Reply:
oh wait, I didn't notice `this._value = value;`

Reply:
yeah, it replaces the value and returns the old

Reply:
then yeah, it makes sense